typescript/src/start_script.ts

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env node
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { execFileSync } from 'child_process';
+import * as path from 'path';
+import { argv } from 'yargs';
+import * as fs from 'fs-extra';
+const fileSystem = require('file-system');
+
+const GOOGLE_GAX_PROTOS_DIR = path.join(
+  __dirname,
+  '..',
+  '..',
+  'node_modules',
+  'google-gax',
+  'protos'
+);
+
+// Add folder of plugin to PATH
+process.env['PATH'] = __dirname + path.delimiter + process.env['PATH'];
+
+let outputDir = '';
+if (argv.output_dir) {
+  outputDir = argv.output_dir as string;
+} else {
+  console.error('Output directory is required: --output_dir path');
+  process.exit(1);
+}
+
+const protoDirs: string[] = [];
+if (argv.I) {
+  if (Array.isArray(argv.I)) {
+    protoDirs.push(...argv.I);
+  } else {
+    protoDirs.push(argv.I as string);
+  }
+}
+const protoDirsArg = protoDirs.map(dir => `-I${dir}`);
+
+const protoFiles = [];
+if (Array.isArray(argv._)) {
+  protoFiles.push(...argv._);
+} else {
+  protoFiles.push(argv._);
+}
+
+// run protoc command to generate client library
+const protocCommand = [
+  `-I${GOOGLE_GAX_PROTOS_DIR}`,
+  `${protoDirsArg}`,
```

Comment:
Wait, this is still incorrect. Just assume there is a space in the path (which is very possible on Windows and sometimes on other systems too).

```
gapic-generator-typescript -I"C:/Documents and Settings/username/protos"
```

so `protoDirsArgs` will become `-IC:/Documents and Settings/username/protos` and, since you convert it to a string in this line, will become separate parameters `-IC:/Documents`, `and`, `Settings/username/protos`.

To fix it, each element of `protoDirsArg` should become an element of `protocCommand` array.